### PR TITLE
refactor(checks-gate): route API-error comment through a template

### PR DESCRIPTION
**Pull request by Kiran01bm**
## What and Why ?

The PR checks gate's API-failure branch built its blocked comment with inline `fmt.Sprintf` and `strings.Contains`, while every other gate (review gate, lock gates, prior-env gate) goes through a `templates.Render…` helper. That left the message inconsistent with its sibling comments (`RenderApplyBlockedByFailingChecks` / `RenderApplyBlockedByInProgressChecks`) and untestable in isolation.

Pure refactor — no behaviour change, no API surface change.

**Changes**
- Add `templates.RenderApplyBlockedByCheckStatusError(env, err)` next to the other gate templates. The function recognises GitHub's `Resource not accessible by integration` permission error and surfaces a targeted hint; all other errors are shown verbatim with a retry instruction.
- Replace the inline rendering in `enforcePassingChecks` with a single call to the new template; drop the now-unused `strings` import.
- Cover the new template with unit tests for generic, permission, and nil-error inputs. Existing handler tests continue to pass — the rendered text is preserved.

NOTE: There are more inline `fmt.Sprintf` pr comments  that could be turned into → `templates.Render` - helps keep things DRY and improves test-ability.

### CODE REVIEW

File: pkg/webhook/templates/apply_commands.go [hunk 1]
```diff
@@ -259,6 +259,33 @@ func RenderApplyBlockedByFailingChecks(environment string, failing []BlockingChe
...
+	}
+
+	sb.WriteString("Unable to verify PR check statuses:\n\n")
+	sb.WriteString("```\n")
+	if err != nil {
+		fmt.Fprintf(&sb, "%s\n", err)
+	}
+	sb.WriteString("```\n\n")
+	fmt.Fprintf(&sb, "Retry:\n```\nschemabot apply -e %s\n```\n", environment)
+
```
[the remark this answers is not in the document]
Comment (Kiran01bm) [reply pkg/webhook/templates/apply_commands.go:284]: fair point - i think we should pivot towards removing the retry block to preserve the prior behavior

**Review by aparajon — APPROVED** [review]
